### ISSUE TITLE
💅 noRedeclare | false positive

### ISSUE THREAD

**Issue by ahmedrowaihi** [biomejs/biome#2372]
### Environment information

```bash
CLI:
  Version:                      1.6.1
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           linux

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "alacritty"
  JS_RUNTIME_VERSION:           "v21.6.0"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "pnpm/8.15.6"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 true

Linter:
  Recommended:                  true
  All:                          false
  Rules:                        complexity/noForEach = "off"
                                suspicious/noExplicitAny = "off"

Workspace:
  Open Documents:
```

### Rule name

suspicious/noRedeclare

### Playground link

https://biomejs.dev/playground/?code=aQBtAHAAbwByAHQAIAB7ACAAYQBsAGwAQQBiAG8AdQB0AHMALAAgAHQAeQBwAGUAIABBAGIAbwB1AHQAIAB9ACAAZgByAG8AbQAgACIAYwBvAG4AdABlAG4AdABsAGEAeQBlAHIALwBnAGUAbgBlAHIAYQB0AGUAZAAiADsACgBpAG0AcABvAHIAdAAgAHQAeQBwAGUAIAB7ACAATQBlAHQAYQBkAGEAdABhACAAfQAgAGYAcgBvAG0AIAAiAG4AZQB4AHQAIgA7AAoACgBmAHUAbgBjAHQAaQBvAG4AIABBAGIAbwB1AHQAQwBhAHIAZAAoAGEAYgBvAHUAdAA6ACAAQQBiAG8AdQB0ACkAIAB7AAoACQByAGUAdAB1AHIAbgAgACgACgAJAAkAPABhAHIAdABpAGMAbABlACAAYwBsAGEAcwBzAE4AYQBtAGUAPQAiAHAAcgBvAHMAZQAgAHAAcgBvAHMAZQAtAHMAbABhAHQAZQAgAG0AYgAtADgAIABkAGEAcgBrADoAcAByAG8AcwBlAC0AaQBuAHYAZQByAHQAIgA%2BAAoACQAJAAkAPABoADIAIABjAGwAYQBzAHMATgBhAG0AZQA9ACIAbQBiAC0AMAAgAHQAZQB4AHQALQAzAHgAbAAgAGYAbwBuAHQALQBzAGUAbQBpAGIAbwBsAGQAIAB0AHIAYQBjAGsAaQBuAGcALQB0AGkAZwBoAHQAIAB0AHIAYQBuAHMAaQB0AGkAbwBuAC0AYwBvAGwAbwByAHMAIAAiAD4ACgAJAAkACQAJAHsAYQBiAG8AdQB0AC4AdABpAHQAbABlAH0ACgAJAAkACQA8AC8AaAAyAD4ACgAJAAkACQB7AC8AKgAgAGIAaQBvAG0AZQAtAGkAZwBuAG8AcgBlACAAbABpAG4AdAAvAHMAZQBjAHUAcgBpAHQAeQAvAG4AbwBEAGEAbgBnAGUAcgBvAHUAcwBsAHkAUwBlAHQASQBuAG4AZQByAEgAdABtAGwAOgAgAGkAbgB0AGUAbgB0AGkAbwBuAGEAbABsAHkAIAAqAC8AfQAKAAkACQAJADwAZABpAHYAIABkAGEAbgBnAGUAcgBvAHUAcwBsAHkAUwBlAHQASQBuAG4AZQByAEgAVABNAEwAPQB7AHsAIABfAF8AaAB0AG0AbAA6ACAAYQBiAG8AdQB0AC4AYgBvAGQAeQAuAGgAdABtAGwAIAB9AH0AIAAvAD4ACgAJAAkAPAAvAGEAcgB0AGkAYwBsAGUAPgAKAAkAKQA7AAoAfQAKAAoAZQB4AHAAbwByAHQAIABjAG8AbgBzAHQAIABtAGUAdABhAGQAYQB0AGEAOgAgAE0AZQB0AGEAZABhAHQAYQAgAD0AIAB7AAoACQB0AGkAdABsAGUAOgAgACIAQQBiAG8AdQB0ACIALAAKAAkAZABlAHMAYwByAGkAcAB0AGkAbwBuADoAIAAiAEwAZQBhAHIAbgAgAGEAYgBvAHUAdAAgAHQAaABlACAAdABlAGMAaAAgAHMAdABhAGMAawAgAGEAbgBkACAAaQBuAHMAcABpAHIAYQB0AGkAbwBuACAAYgBlAGgAaQBuAGQAIABDAGgAYQBkAE4AZQB4AHQALgAiACwACgB9ADsACgAKAGUAeABwAG8AcgB0ACAAZABlAGYAYQB1AGwAdAAgAGYAdQBuAGMAdABpAG8AbgAgAEEAYgBvAHUAdAAoACkAIAB7AAoACQByAGUAdAB1AHIAbgAgACgACgAJAAkAPABkAGkAdgAgAGMAbABhAHMAcwBOAGEAbQBlAD0AIgBjAG8AbgB0AGEAaQBuAGUAcgAgAG0AaQBuAC0AaAAtAHMAYwByAGUAZQBuACAAcAB5AC0AOAAiAD4ACgAJAAkACQA8AGgAMQAgAGMAbABhAHMAcwBOAGEAbQBlAD0AIgB0AGUAeAB0AC0ANAB4AGwAIABmAG8AbgB0AC0AYgBvAGwAZAAgAHQAcgBhAGMAawBpAG4AZwAtAHQAaQBnAGgAdAAgAGwAZwA6AHQAZQB4AHQALQA1AHgAbAAiAD4ACgAJAAkACQAJAEEAYgBvAHUAdAAKAAkACQAJADwALwBoADEAPgAKAAkACQAJADwAcAAgAGMAbABhAHMAcwBOAGEAbQBlAD0AIgBtAGIALQAxADAAIABtAHQALQAyAC4ANQAgAHQAZQB4AHQALQB4AGwAIAB0AGUAeAB0AC0AbQB1AHQAZQBkAC0AZgBvAHIAZQBnAHIAbwB1AG4AZAAiAD4ACgAJAAkACQAJAEwAZQBhAHIAbgAgAGEAYgBvAHUAdAAgAHQAaABlACAAdABlAGMAaAAgAHMAdABhAGMAawAgAGEAbgBkACAAaQBuAHMAcABpAHIAYQB0AGkAbwBuACAAYgBlAGgAaQBuAGQAIABDAGgAYQBkAE4AZQB4AHQALgAKAAkACQAJADwALwBwAD4ACgAJAAkACQA8AGQAaQB2ACAAYwBsAGEAcwBzAE4AYQBtAGUAPQAiAHMAcABhAGMAZQAtAHkALQAxADAAIgA%2BAAoACQAJAAkACQB7AGEAbABsAEEAYgBvAHUAdABzAC4AbQBhAHAAKAAoAHAAKQAgAD0APgAgACgACgAJAAkACQAJAAkAPABBAGIAbwB1AHQAQwBhAHIAZAAgAGsAZQB5AD0AewBwAC4AXwBpAGQAfQAgAHsALgAuAC4AcAB9ACAALwA%2BAAoACQAJAAkACQApACkAfQAKAAkACQAJADwALwBkAGkAdgA%2BAAoACQAJADwALwBkAGkAdgA%2BAAoACQApADsACgB9AAoA

### Expected result

It should not throw an error

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct
